core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -703,8 +703,8 @@ class KafkaServer(
                 case None =>
                   info(s"Broker registration for controller $controllerId is not available in the metadata cache")
               }
-            case None =>
-              info("No controller present in the metadata cache")
+            case _ =>
```

Comment:
this one confuses me. Isn't `case None` clearer than `case _` ?

Reply:
The condition covers more than None. It includes Some(controller: KRaftCachedControllerId) as well.

Reply:
Let me add more explicit condition to prevent this confusion.